widget/check.go

```diff
@@ -167,42 +93,65 @@ func (c *Check) Hide() {
 }
 
 // MouseIn is called when a desktop pointer enters the widget
-func (c *Check) MouseIn(*desktop.MouseEvent) {
-	if c.Disabled() {
-		return
-	}
-	c.hovered = true
-	c.Refresh()
+func (c *Check) MouseIn(me *desktop.MouseEvent) {
+	c.MouseMoved(me)
 }
 
 // MouseOut is called when a desktop pointer exits the widget
 func (c *Check) MouseOut() {
-	c.hovered = false
-	c.Refresh()
+	if c.hovered {
+		c.hovered = false
+		c.Refresh()
+	}
 }
 
 // MouseMoved is called when a desktop pointer hovers over the widget
-func (c *Check) MouseMoved(*desktop.MouseEvent) {
+func (c *Check) MouseMoved(me *desktop.MouseEvent) {
+	if c.Disabled() {
+		return
+	}
+
+	oldHovered := c.hovered
+	if c.minSize.Width == 0 && c.minSize.Height == 0 {
+		c.hovered = true
+	} else if me.Position.X <= c.minSize.Width && me.Position.Y <= c.minSize.Height {
+		c.hovered = true
+	} else {
+		// mouse outside the active area of the widget
+		c.hovered = false
+	}
```

Comment:
You can simplify the code and make it branch-free like the code example below. I think it also is a bit more readable perhaps?
```go
	c.hovered = c.minSize.IsZero() || (me.Position.X <= c.minSize.Width && me.Position.Y <= c.minSize.Height)
```

Reply:
Nice one